ec2/ec2.go

```diff
@@ -138,20 +139,21 @@ func (ec2 *EC2) query(params map[string]string, resp interface{}) error {
 	}
 	sign(ec2.Auth, "GET", endpoint.Path, params, endpoint.Host)
 	endpoint.RawQuery = multimap(params).Encode()
-	if debug {
-		log.Printf("get { %v } -> {\n", endpoint.String())
-	}
+	log.Debugf("GET %v", endpoint.String())
 	r, err := ec2.httpClient.Get(endpoint.String())
 	if err != nil {
 		return err
 	}
 	defer r.Body.Close()
 
-	if debug {
-		dump, _ := httputil.DumpResponse(r, true)
-		log.Printf("response:\n")
-		log.Printf("%v\n}\n", string(dump))
+	dump := []byte{}
+	if tracingEnabled() {
+		dump, _ = httputil.DumpResponse(r, true)
+	} else {
+		dump, _ = httputil.DumpResponse(r, false)
```

Comment:
As appose to before, this will always get called if you are not debugging or tracing. Is this line expensive?